### ISSUE TITLE
Splunk Operator: helm chart can't be installed if there is not pvc being created

### ISSUE THREAD

**Issue by iaroslav-nakonechnikov** [splunk/splunk-operator#1024]
### Please select the type of request

Bug

### Tell us more

**Describe the request**
I've noticed, that helm chart failing with installation, when it was requested to use ephemeral volumes. 

**Expected behavior**
splunk-operator chart works as expected by documenation. 

**Splunk setup on K8S**
eks 1.22
splunk-operator 2.1.1

**Reproduction/Testing steps**
install splunk-operator with:
```
volumes:
  - name: app-staging
    hostPath:
      path: /some/host/path
      type: DirectoryOrCreate
volumeMounts:
  - mountPath: /opt/splunk/appframework/
    name: app-staging
```
or
```
volumes:
  - name: app-staging
    emptyDir:
      sizeLimit: 10Gi
volumeMounts:
  - mountPath: /opt/splunk/appframework/
    name: app-staging
```

**Additional context(optional)**
why it is needed?

1. for testing/development purposes - it starts for 15sec. 
2. for making it independent from node az. By default pvc is connected to az, so need add configs to make it works. 
with letting to use hostPath - it becoming independent.
3. i see no reason to stick to permanent storage. Maybe best scenario will be to use s3 bucket for that volume.

**Comment by iaroslav-nakonechnikov** [comment]
```Error: rendered manifests contain a resource that already exists. Unable to continue with install: PersistentVolumeClaim "splunk-operator-app-download" in namespace "splunk-operator" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "splunk-operator"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "splunk-operator"```

looks like something hardcoded in chart.

**Comment by vivekr-splunk** [comment]
please use 2 step process now to install operator and install splunk instance, we are working on fixing few issues in splunk operator helm chart. here is the example 
```
 helm install splunk-operator splunk/splunk-operator --create-namespace -n splunk-operator --set splunkOperator.clusterWideAccess=false
NAME: splunk-operator
LAST DEPLOYED: Mon Jan  9 09:51:20 2023
NAMESPACE: splunk-operator
STATUS: deployed
REVISION: 1
TEST SUITE: None


 helm install --set sva.s1.enabled=true --set splunk-operator.enabled=false splunk-s1 splunk/splunk-enterprise -n splunk-operator --create-namespace 
NAME: splunk-s1
LAST DEPLOYED: Mon Jan  9 09:51:46 2023
NAMESPACE: splunk-operator
STATUS: deployed
REVISION: 1
TEST SUITE: None

 kubectl get pods -n splunk-operator  -w
NAME                                                  READY   STATUS    RESTARTS   AGE
splunk-operator-controller-manager-7578cd4c87-87mvj   2/2     Running   0          108s
splunk-stdln-standalone-0                             1/1     Running   0          86s
```

if you want to use ephemeral volume for operator also you can change them in values.yaml file. 

```
  # Specify volumes for Splunk Operator pod, append additional volumes to list
  # reference: https://kubernetes.io/docs/concepts/storage/volumes/
  volumes:
  - name: app-staging
    persistentVolumeClaim:
      claimName: splunk-operator-app-download
# - name:
#   persistentVolumeClaim:
#     claimName:

  # Specify volume mounts for the manager container, append additional volume mounts to list
  # reference: https://kubernetes.io/docs/tasks/configure-pod-container/configure-volume-storage/
  volumeMounts:
  - mountPath: /opt/splunk/appframework/
    name: app-staging
# - mountPath:
#   name:
```

**Comment by iaroslav-nakonechnikov** [comment]
i see that just installing `splunk-operator` works as expected.  
i don't see any difference with `splunk-enterprise`. 
same config which was working on version 2.0.0 - works in 2.1.1.
What about chart - i removed `app_download.yaml` from templates, and it was easy thing to make it work as expected as well. 


and as we are using terraform, so here is how it is installed atm:
```
resource "helm_release" "splunk-operator" {
  name = "splunk-operator"
  # as chart is broken, we've put it in repo for some time.
  chart = "files/splunk-operator-1.0.0.tgz"
  namespace        = "splunk-operator"
  create_namespace = true
  wait_for_jobs    = true
  depends_on       = [module.eks, module.load_balancer_controller_irsa, null_resource.create_kube_config]
  cleanup_on_fail  = true
  force_update     = true
  replace          = true
  verify           = false

  dynamic "set" {
    for_each = {
      "image.repository"               = "internal.proxy/docker/splunk/splunk:9.0.3"
      "kubeRbacProxy.image.repository" = "internal.proxy/gcr.io-docker-proxy/kubebuilder/kube-rbac-proxy"
      # in case we will need to have newer image for rbac-proxy
      # "kubeRbacProxy.image.tag"                = "v0.13.0"
      "splunkOperator.image.repository"          = "internal.proxy/docker/splunk/splunk-operator:2.1.1"
      "splunkOperator.tolerations[0].key"        = "function"
      "splunkOperator.tolerations[0].operator"   = "Equal"
      "splunkOperator.tolerations[0].value"      = "splunk-operator"
      "splunkOperator.tolerations[0].effect"     = "NoSchedule"
      "splunkOperator.volumes[0].name"           = "app-staging"
      "splunkOperator.volumes[0].hostPath.path"  = "/mnt/scratch/app-staging"
      "splunkOperator.volumes[0].hostPath.type"  = "DirectoryOrCreate"
      "splunkOperator.volumeMounts[0].mountPath" = "/opt/splunk/appframework/"
      "splunkOperator.volumeMounts[0].name"      = "app-staging"
    }
    content {
      name  = set.key
      value = set.value
    }
  }
}
```

**Comment by vivekr-splunk** [comment]
hello @iaroslav-nakonechnikov  thanks for your response, I am not sure if I understand properly , for ephemeral volume i tested today for splunk-operator and it works fine. and I think as for your reply it has worked for you also. for enterprise splunk , for testing purpose try setting `storageclass: manual` in splunk-entrprise values.yaml file.  currently dependency between splunk-enterprise chart and splunk-operator chart is not working , we are aware of the issue and will be fixed in upcoming release. until then please install the charts separately as mentioned above.

**Comment by iaroslav-nakonechnikov** [comment]
@vivekr-splunk i don't want `manual`, i expect that configuration will work as it is described in documentation.

**Comment by vivekr-splunk** [comment]
i have fixed issue of creating pvc even when ephemeral volumes is used. https://github.com/splunk/splunk-operator/pull/1037. let me know if you have any other issue regarding persistent volume.

**Comment by iaroslav-nakonechnikov** [comment]
Super! 
waiting till release will be released, so it would be possible to test in correct way.

**Comment by iaroslav-nakonechnikov** [comment]
yes! it solved with 2.2.0 release. Thanks!
